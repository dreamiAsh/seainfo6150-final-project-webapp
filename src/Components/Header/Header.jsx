import React from "react";
import styles from "./Header.module.css";
import { Link, BrowserRouter } from "react-router-dom";

const Header = (props) => {
    return (
        <div className={styles.box}>
            <div className={styles.inputBox}>
                <input className={styles.input} placeholder=" Find your recipe"/>
                <BrowserRouter forceRefresh={true}><Link to="/search" className={styles.Go}>GO</Link></BrowserRouter>
            </div>
            <p className={styles.title}>Home recipe</p>
            <BrowserRouter forceRefresh={true}><Link to="/Signin" className={styles.signIn}>Sign in</Link></BrowserRouter>
        </div>
    );
};

export default Header;
