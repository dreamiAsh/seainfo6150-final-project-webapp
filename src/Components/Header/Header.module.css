*{margin: 0;padding: 0;list-style: none;font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;}
.box{
    width: 100%;
    height: 80px;
    background: #DC7E20;
    display: grid;
    flex-flow: column-reverse;
    justify-content: space-around;
    align-items: stretch;
    grid-template-columns: repeat(3, 30fr);
    text-align: center;
}
.inputBox{
    width: 80%;
    margin: 25px 0 0 35px;
}
.input{
    height: 25px;
    width: 55%;
    line-height: 80px;
    float: left;
}
.Go{
    height: 28px;
    width: 34px;
    background: #eee;
    float: left;
    color: #C6582B;
    line-height: 28px;
    text-align: center;
    font-weight: bold;
    font-size: 14px;
    text-decoration: none;
}
.title{
    height: 80px;
    font-size: 35px;
    color: white;
    line-height: 80px;
    font-weight: bold;
    font-family: Courier New;
}
.signIn{
    height: 80px;
    font-size: 16px;
    color: white;
    line-height: 80px;
    font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
}