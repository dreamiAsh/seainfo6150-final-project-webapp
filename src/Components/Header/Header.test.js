import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Header from "./Header.jsx";

it("renders the Header component correctly", () => {
  const { container } = render(<Header />);
  expect(container).toMatchSnapshot();
});