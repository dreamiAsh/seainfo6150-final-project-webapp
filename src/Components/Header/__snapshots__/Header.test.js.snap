// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Header component correctly 1`] = `
<div>
  <div
    class="box"
  >
    <div
      class="inputBox"
    >
      <input
        class="input"
        placeholder=" Find your recipe"
      />
      <a
        class="Go"
        href="/search"
      >
        GO
      </a>
    </div>
    <p
      class="title"
    >
      Home recipe
    </p>
    <a
      class="signIn"
      href="/signin"
    >
      Sign in
    </a>
  </div>
</div>
`;
